addons/project/models/project.py

```diff
@@ -1217,10 +1320,72 @@ def default_get(self, default_fields):
 
         return vals
 
+    def read(self, fields=None, load='_classic_read'):
+        if fields and self.env.user.has_group('base.group_portal'):
+            unauthorized_fields = set(fields) - self.SELF_READABLE_FIELDS
+            if unauthorized_fields:
+                raise AccessError(_('You cannot read %s field in task.', ', '.join(unauthorized_fields)))
```

Comment:
Maybe this could be extracted to a method given it's repeated half a dozen times?